src/upchunk.ts

```diff
@@ -1,9 +1,169 @@
 import { EventTarget, Event } from 'event-target-shim';
-import xhr, { XhrUrlConfig, XhrHeaders, XhrResponse } from 'xhr';
+import xhr, { type XhrUrlConfig, type XhrHeaders, type XhrResponse } from 'xhr';
+
+const DEFAULT_CHUNK_SIZE = 30720;
+const DEFAULT_MAX_CHUNK_SIZE = 512000; // in kB
```

Comment:
tbh I'm not sure but I think that can happen (since there are tradeoffs in overall time between big chunk that takes awhile and several smaller chunks that each take shorter but add overhead on total RTT). This is just keeping it original value (originally defined inline in `UpChunk::constructor()`

Reply:
There's a simple algorithm on dynamic chunk sizing at the moment: the chunk size doubles when the last chunk duration (`lastChunkInterval`) was less than 10 seconds, and it halves when the last chunk took more than 30 seconds to upload. So as-is we should only see a 512MB chunk if the client was just able to upload a 256MB chunk in < 10 seconds (25.6 MB/s / 205 Mbit/s).

If we don't scale the chunk size up, then high bandwidth clients are kneecapped by round trip time and don't really have the opportunity to hit full throughput. However there's of course a tradeoff where it starts using too much memory.

Honestly 512MB is as large or larger than what I saw from any of the other sites I tested back in February. Google Drive, for example, caps out at 200MB. If there's concern that 512MB is too large, then 256MB is probably a more reasonable limit.

I also worry that 256 KB is too low, but then nobody will ever hit that unless it's taking them >30 seconds to upload 512KB. Really this setting is about trying to balance upload throughput vs memory usage vs maximum amount of time lost if a chunk has to retry.